Entities/Natural/Trees/ThinLogsOnDeath.as

```diff
@@ -0,0 +1,43 @@
+//tree making logs on death script
+
+#include "TreeCommon.as"
+
+void onDie(CBlob@ this)
+{
+	Vec2f pos = this.getPosition();
+	f32 fall_angle = 0.0f;
+
+	if (this.exists("tree_fall_angle"))
+	{
+		fall_angle = this.get_f32("tree_fall_angle");
+	}
+
+	TreeSegment[]@ segments;
+	this.get("TreeSegments", @segments);
+	if (segments is null)
+		return;
+
+	for (uint i = 0; i < (segments.length); i++)
+	{
+		TreeSegment@ segment = segments[i];
+
+		if (getNet().isServer())
+		{
+			pos = this.getPosition() + (segment.start_pos + segment.end_pos) / 2.0f;
+			pos.y -= 4.0f; // TODO: fix logs spawning in ground
+			CBlob@ log = server_CreateBlob("log_thin", this.getTeamNum(), pos);

```

Comment:
You don't need an near-identical file to `LogsOnDeath.as` just to spawn thin logs

Reply:
To elaborate, `LogsOnDeath.as` makes use of the `onDie(CBlob@ this)` hook, as such you can make clever use of `getName();` ;)